Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1466,8 +1466,13 @@ bool WebProcessProxy::wasPreviouslyApprovedFileURL(const URL& url) const
     return m_previouslyApprovedFilePaths.contains(fileSystemPath);
 }
 
-void WebProcessProxy::recordUserGestureAuthorizationToken(PageIdentifier pageID, WTF::UUID authorizationToken)
+void WebProcessProxy::recordUserGestureAuthorizationToken(FrameIdentifier frameID, PageIdentifier pageID, WTF::UUID authorizationToken)
 {
+    if (RefPtr dataStore = websiteDataStore()) {
+        if (RefPtr frame = WebFrameProxy::webFrame(frameID))
```

Comment:
I think we only want to call `didHaveUserInteractionForSiteIsolation` if the frame is a main frame or if the target frame's site matches the main site right? Isn't the ITP heuristic "domains you've interacted with as a *first party* only"? We don't care about interactions with third party frames for this heuristic.

Reply:
Oh, that's a good point. Asking...